frontend/model/contracts/group.js

```diff
@@ -703,13 +721,28 @@ sbp('chelonia/defineContract', {
           { username: meta.username, dateLeft: meta.createdDate },
           { meta, state, getters }
         )
+
+        meta.isRemoveOurselves = true
         sbp('gi.contracts/group/pushSideEffect', contractID,
           ['gi.contracts/group/removeMember/sideEffect', {
             meta,
             data: { member: meta.username, reason: data.reason || '' },
             contractID
           }]
         )
+      },
+      sideEffect ({ meta, contractID }, { state }) {
+        const { loggedIn } = sbp('state/vuex/state')
+        const { profiles = {} } = state
+        const myProfile = profiles[loggedIn.username] || null
+
+        if (loggedIn.username !== meta.username &&
+          isContractYoungerThanUser(meta, myProfile)) {
+          sbp('gi.notifications/emit', 'MEMBER_LEFT', { // emit a notification for a member addition.
+            groupID: contractID,
+            username: meta.username
+          })
+        }
```

Comment:
You should be able to handle everything in the `removeMember` sideEffect.

So this function doesn't need to be created.

To check for the case where the member is removing themselves from the group, check for the condition where `data.member === meta.username && data.member === loggedIn.username`. Those 3 will equal each other in such instances.